src/reducers/index.js

```diff
@@ -21,84 +21,86 @@ import convertColumnElementsToData from 'convertColumnElementsToData';
 import pick from 'lodash/pick';
 import shallowEqual from 'shallowEqual';
 
-/**
- * Input state set from props
- */
-const DEFAULT_INPUT_STATE = {
-  columnProps: [],
-  columnGroupProps: [],
-  elementTemplates: {
-    cell: [],
-    footer: [],
-    groupHeader: [],
-    header: [],
-  },
-  elementHeights: {
-    footerHeight: 0,
-    groupHeaderHeight: 0,
-    headerHeight: 0,
-  },
-  rowSettings: {
-    bufferRowCount: undefined,
-    rowHeight: 0,
-    rowHeightGetter: () => 0,
-    rowsCount: 0,
-    subRowHeight: 0,
-    subRowHeightGetter: () => 0,
-  },
-  scrollFlags: {
-    overflowX: 'auto',
-    overflowY: 'auto',
-    showScrollbarX: true,
-    showScrollbarY: true,
-  },
-  tableSize: {
-    height: undefined,
-    maxHeight: 0,
-    ownerHeight: undefined,
-    useMaxHeight: false,
-    width: 0,
-  },
-};
-
-/**
- * Output state passed as props to the the rendered FixedDataTable
- * NOTE (jordan) rows may contain undefineds if we don't need all the buffer positions
- */
-const DEFAULT_OUTPUT_STATE = {
-  columnReorderingData: {},
-  columnResizingData: {},
-  firstRowIndex: 0,
-  firstRowOffset: 0,
-  isColumnReordering: false,
-  isColumnResizing: false,
-  maxScrollX: 0,
-  maxScrollY: 0,
-  rowHeights: {},
-  rows: [], // rowsToRender
-  scrollContentHeight: 0,
-  scrollX: 0,
-  scrollY: 0,
-};
+function getInitialState() {
```

Comment:
Internals here are unchanged.  Just moved the consts within the function.

Reply:
Would it be better to assign getInitialState to a function defined within a function which also contains the immutable constants, so that they aren't recalculated, but simply reused for every function call.

something like: 
```javascript
const getInitialState = (function() {
    const DEFAULT_INPUT_STATE = {...};
    const DEFAULT_OUTPUT_STATE = {...};
    const DEFAULT_INTERNAL_STATE = {...};
    const INITIAL_STATE = Object.assign({},
        DEFAULT_INPUT_STATE,
        DEFAULT_OUTPUT_STATE,
        DEFAULT_INTERNAL_STATE
    );
    return function() { return INITIAL_STATE };
 })();
```

Reply:
Wouldn't that lead to the same bug where each call to getInitialState returned the same object?

Maybe the confusion is I should update the variable names so their no longer all caps, since these aren't global consts anymore?

Reply:
Ah yes it will, my bad. 
In that case, adding a cloneDeep should suffice.
```javascript
const getInitialState = (function() {
    const DEFAULT_INPUT_STATE = {...};
    const DEFAULT_OUTPUT_STATE = {...};
    const DEFAULT_INTERNAL_STATE = {...};
    const INITIAL_STATE = Object.assign({},
        DEFAULT_INPUT_STATE,
        DEFAULT_OUTPUT_STATE,
        DEFAULT_INTERNAL_STATE
    );
    return function() { return _.cloneDeep(INITIAL_STATE) };
 })();
```

Not sure if there would be any performance benefits, since the function is just called N times for N instances (if I'm right...)

and yes, lowercase seems better.

Reply:
Will update the variable names to be camelCase.

I don't follow the cloneDeep suggestion.  Would that provide any advantage over redeclaring?  It seems more complex.

Reply:
I just wanted to get rid of redefining but yea this seems more complex...

Btw, what about just having a cloneDeep here instead.
```javascript
function reducers(state = cloneDeep(getInitialState()), action) {
    ....
}
```
This way, there won't be any change in other lines.

Reply:
Yes, that would have the same effect.  It wouldn't work over the IntegerBufferSet, but that could be pulled out.

That said, I don't think it's worth it as a solution since cloneDeep doesn't work on all arguments, and the benefit would be just minimizing this changeset.

Reply:
Ah, understood and agreed. Thanks for the explanation.